spring-cloud-stream/src/test/java/org/springframework/cloud/stream/binder/PollableConsumerTests.java

```diff
@@ -414,6 +415,43 @@ public void testRequeue() {
 		verify(callback).acknowledge(Status.REQUEUE);
 	}
 
+	@Test
+	public void testRequeueWithNoAcknowledgementCallback() {
+		TestChannelBinder binder = createBinder();
+		MessageConverterConfigurer configurer = this.context
+				.getBean(MessageConverterConfigurer.class);
+
+		DefaultPollableMessageSource pollableSource = new DefaultPollableMessageSource(
+				this.messageConverter);
+		configurer.configurePolledMessageSource(pollableSource, "foo");
+		AcknowledgmentCallback callback = mock(AcknowledgmentCallback.class);
+		pollableSource.addInterceptor(new ChannelInterceptor() {
+
+			@Override
+			public Message<?> preSend(Message<?> message, MessageChannel channel) {
+				return MessageBuilder.fromMessage(message)
+						.build();
+			}
+
+		});
+		ExtendedConsumerProperties<Object> properties = new ExtendedConsumerProperties<>(null);
+		properties.setMaxAttempts(2);
+		properties.setBackOffInitialInterval(0);
+		binder.bindPollableConsumer("foo", "bar", pollableSource, properties);
+		final AtomicInteger count = new AtomicInteger();
+		try {
+			assertThat(pollableSource.poll(received -> {
+				count.incrementAndGet();
+				throw new RequeueCurrentMessageException("test retry");
+			})).isTrue();
+		}
+		catch (Exception e) {
+			// no op
+		}
```

Comment:
This test passes when the DPMS is reverted (ignores the NPE). Shouldn't we just remove the try/catch?